src/focus-camera/focus-first-residue.ts

```diff
@@ -60,12 +60,44 @@ export class FocusFirstResidue implements FocusFactoryI {
             imageRender.canvas3d.camera.setState(state);
         };
     }
-    getAxesToFlip(positions: Float32Array, origin: Vec3, up: Vec3, normalDir: Vec3) {
-        const toYAxis = calculateDisplacement(positions, origin, normalDir);
-        const toXAxis = calculateDisplacement(positions, origin, up);
+    getAxesToFlip(position: Vec3, origin: Vec3, up: Vec3, normalDir: Vec3) {
+        const toYAxis = calculateDisplacement(position, origin, normalDir);
+        const toXAxis = calculateDisplacement(position, origin, up);
         const Axes: string[] = [];
-        if (toYAxis[0] < 0) Axes.push('aroundY');
-        if (toXAxis[0] < 0) Axes.push('aroundX');
+        if (toYAxis < 0) Axes.push('aroundY');
+        if (toXAxis < 0) Axes.push('aroundX');
         return Axes;
     }
+}
+
+function getFirstResidueOrAveragePosition(structure: Structure, caPositions: Float32Array): Vec3 {
+    //TODO Is this the best way to test single chain?
```

Comment:
Chains can be split into multiple units if they are too big too. But in that case, this is probably a good enough approximation.